### DISCUSSION TITLE
fabric.Control 创建的自定义控件  第一次选中不显示

### DISCUSSION THREAD

**Discussion by list12318** [fabricjs/fabric.js#9797] [Q&A]
1、我使用new fabric.Control() 创建了我的自定义删除按钮
2、每一次初始化渲染很多多边形 polygon 后 使用setActiveObject 选中某个polygon后，自定义删除控件并不显示，当我拖动或者切换选中另一个polygon后才显示
3、请问我该如何修改代码让他第一次选中就可以显示删除控件呢？
4、我使用的是onMouseDown 判断 polygon 是否选中
以下是我的代码截图：

Automated translation
```
1. I created my custom delete button using new fabric.Control()
2. After each initialization and rendering of many polygons, I use setActiveObject to select a polygon. The custom delete control does not display. It only displays when I drag or switch to select another polygon.
3. How can I modify the code so that the delete control can be displayed when it is selected for the first time?
4. I use onMouseDown to determine whether the polygon is selected. The following is a screenshot of my code:
```

![image](https://github.com/fabricjs/fabric.js/assets/33713614/76048900-7b25-49d1-b2b3-672b3db987f8)

![image](https://github.com/fabricjs/fabric.js/assets/33713614/577a8a4f-9c97-45b4-a1b0-1488c1454aec)

![image](https://github.com/fabricjs/fabric.js/assets/33713614/365a52b6-bf88-46ff-b78a-00059eca639d)

**Comment by asturur** [comment]
hi @list12318 please try to post in english.
I m attaching a translation of your question to your message

**Reply by list12318** [reply]
Thank you, your translation is basically accurate. I will explain the second point to you again

2. When initializing rendering many polygons using the foreach method, the first time a polygon is selected through the setActiveObject () method, the custom control will not be displayed unless you drag or switch to another polygon

My English is not very good, I'm sorry

**Reply by list12318** [reply]
You can definitely look at this issue in your free time, thank you
